scripts/fuzz_opt.py

```diff
@@ -164,23 +164,66 @@ def randomize_fuzz_settings():
     print('randomized settings (NaNs, OOB, legalize):', NANS, OOB, LEGALIZE)
 
 
-IMPORTANT_INITIAL_CONTENTS = [
-    # Perenially-important passes
-    os.path.join('lit', 'passes', 'optimize-instructions.wast'),
-    os.path.join('passes', 'optimize-instructions_fuzz-exec.wast'),
-
-    # Recently-added or modified passes. These can be added to and pruned
-    # frequently.
-    os.path.join('lit', 'passes', 'once-reduction.wast'),
-    os.path.join('passes', 'remove-unused-brs_enable-multivalue.wast'),
-    os.path.join('lit', 'passes', 'optimize-instructions-bulk-memory.wast'),
-    os.path.join('lit', 'passes', 'optimize-instructions-ignore-traps.wast'),
-    os.path.join('lit', 'passes', 'optimize-instructions-gc.wast'),
-    os.path.join('lit', 'passes', 'optimize-instructions-gc-iit.wast'),
-    os.path.join('lit', 'passes', 'optimize-instructions-call_ref.wast'),
-    os.path.join('lit', 'passes', 'inlining_splitting.wast'),
-]
-IMPORTANT_INITIAL_CONTENTS = [os.path.join(shared.get_test_dir('.'), t) for t in IMPORTANT_INITIAL_CONTENTS]
+def get_important_initial_contents():
+    FIXED_IMPORTANT_INITIAL_CONTENTS = [
+        # Perenially-important passes
+        os.path.join('lit', 'passes', 'optimize-instructions.wast'),
+        os.path.join('passes', 'optimize-instructions_fuzz-exec.wast'),
+    ]
+    MANUAL_RECENT_INITIAL_CONTENTS = [
+        # Recently-added or modified passes. These can be added to and pruned
+        # frequently.
+        os.path.join('lit', 'passes', 'once-reduction.wast'),
+        os.path.join('passes', 'remove-unused-brs_enable-multivalue.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-bulk-memory.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-ignore-traps.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-gc.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-gc-iit.wast'),
+        os.path.join('lit', 'passes', 'optimize-instructions-call_ref.wast'),
+        os.path.join('lit', 'passes', 'inlining_splitting.wast'),
+    ]
+    RECENT_DAYS = 30
+
+    def auto_select_recent_initial_contents():
+        # Returns the list of test wast/wat files added or modified within the
+        # RECENT_DAYS number of days
+        # 1. Print 'git log' with changed file status and without commit message,
+        #    with commits within RECENT_DAYS number of days
+        # 2. Pick up lines in the form of
+        #    A       test/../something.wast
+        #    M       test/../something.wast
+        #    (wat extension is also included)
+        p = re.compile(r'^[AM]\stest' + os.sep + r'(.*\.(wat|wast))$')
+        log = run(['git', 'log', '--name-status', '--format=', '--date=relative', '--no-renames', f'--since="{RECENT_DAYS} days ago"'], silent=True).splitlines()
```

Comment:
My git knowledge about time-based queries is rusty - will this look at `RECENT_DAYS` before the commit (or authored?) date of `HEAD` or at `RECENT_DAYS` since the current system time? I hope it's the former so that issues can be reproduced even months later by checking out the original commit.

Reply:
> will this look at `RECENT_DAYS` before the commit (or authored?)

Before the commit.

> date of `HEAD` or at `RECENT_DAYS` since the current system time?

The current system time.

Reply:
Wait, so is it before the commit or before the current system time?

Reply:
Sorry I maybe misunderstood your question..?

So this only considers the commit time. So if you authored a commit 6 months ago but committed it 3 days ago, it will show up. Also "%d days ago" is compared to the current time, not the time of `HEAD`. So if you run this `git log` command on a local git repo you checked out more than a month ago and haven't touched, it will turn up nothing.

Reply:
Got it, thanks. Would it be possible to make it compare to the time of `HEAD` instead? That way the fuzzer will always behave the same way for a given commit and seed no matter when it is run. Otherwise I think we'll have the same reproducibility and determinism issues that previous solutions for this had.

Reply:
Oh, that's a very good point... Yeah, the result of the specific version(commit) of the fuzzer for a given seed shouldn't change. Will fix.